templates/common/infra/bicep/core/security/keyvault.bicep

```diff
@@ -12,6 +12,8 @@ resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
   properties: {
     tenantId: subscription().tenantId
     sku: { family: 'A', name: 'standard' }
+    enablePurgeProtection: true
```

Comment:
FWIW - We added explicit support to `azd` to purge key vaults (upon user confirmation) to `azd down` because we wanted to support the case where a user runs `azd down` to delete their app and then `azd up` again to recreate it "from scratch".

I'm a little nervous about this change causing us to end up in a state where we end up burning a keyvault name until the underlying resource is purged.

Makes sense for us to enable soft delete here as it looks like this time next year it will be forced on for everyone: https://learn.microsoft.com/en-us/azure/key-vault/general/soft-delete-change regardless of what the bicep says.

Reply:
I currently have 30 security alerts that say "Key Vaults must have Soft Delete and Purge Protection enabled", so I'm trying to figure how I'd address them without this change. So developers would need to slightly tweak their key vault name between a down and up, right? You could put a warning in the purge confirmations about needing to change key vault name, perhaps? Hmm.